includes/User.php

```diff
@@ -4842,4 +4842,23 @@ public static function getRightDescription( $right ) {
 		$msg = wfMessage( $key );
 		return $msg->isBlank() ? $right : $msg->text();
 	}
+
+
+	/**
+	 * We want to use one source for username.
+	 * This function will perform the lookup if
+     * $wgLookupUsername is true
+	 *
+	 * @param $userId int userId
+	 * @param $name string anon username
+	 * @return string
+	 */
+	public static function getUsername( $userId, $name ) {
+		global $wgLookupUsername;
+		// if anon or we don't want to do a lookup, return the provided name
+		if( $userId == 0 || !$wgLookupUsername ){
+			return $name;
+		}
+		return self::whoIs($userId);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)